src/v/cloud_topics/level_one/compaction/source.cc

```diff
@@ -122,103 +176,114 @@ ss::future<ss::stop_iteration> compaction_source::map_building_iteration() {
         co_return ss::stop_iteration::yes;
     }
 
-    if (_map_building_it == _dirty_range_intervals.cend()) {
+    if (_dirty_range_it == _dirty_range_intervals.crend()) {
         co_return ss::stop_iteration::yes;
     }
 
-    const auto& dirty_range = *_map_building_it;
-    const auto& start_offset = dirty_range.base_offset;
-    const auto& max_offset = dirty_range.last_offset;
-
-    cloud_topic_log_reader_config config(start_offset, max_offset, _as);
-    auto rdr = model::record_batch_reader(
-      std::make_unique<level_one_log_reader_impl>(
-        config, _ntp, _tp, _metastore, _io));
-
-    auto res = co_await std::move(rdr).consume(
-      map_building_reducer(*_map), model::no_timeout);
-
-    std::optional<kafka::offset> min_offset_indexed;
-    std::optional<kafka::offset> max_offset_indexed;
-    ss::stop_iteration ret;
-
-    if (res.map_is_full) {
-        // Rely on the result from the reducer to set min/max offset indexed.
-        // Stop iterating since map is full.
-        min_offset_indexed = res.min_offset_indexed;
-        max_offset_indexed = res.max_offset_indexed;
-        ret = ss::stop_iteration::yes;
-    } else {
-        // The entire range was indexed. We can continue iterating until the map
-        // is full or we are preempted.
-        min_offset_indexed = start_offset;
-        max_offset_indexed = max_offset;
-        ret = ss::stop_iteration::no;
+    const auto& dirty_range = *_dirty_range_it;
+
+    auto dirty_range_intervals = intervals_for_dirty_range(
+      dirty_range, _extents);
```

Comment:
Let's call this something else. It is very easy to confuse this with _dirty_range_intervals. How about extent_aligned_ranges or something